### ISSUE TITLE
com_finetune_weights file

### ISSUE THREAD

**Issue by yuan0821** [spoonsso/dannce#49]
Hello Timothy. May I please ask you to provide some explain about the file used for fine-tune? Below is my steps for your demo data. Thank you very much if you could give your comments. 

1. com-train ../../configs/com_mouse_config.yaml
com_finetune_weights: ??? 
Q1: Does COM net use finetune train_model? 
Q2: if so, where can I find the pre-trained weight files for **mouse** COM trained model?
Q3: is it ok if i use your pre-trained mouse weight file for our lab mouse data training?

2. com-predict ../../configs/com_mouse_config.yaml
Q4: Does COM use pre-trained COM finetune weights file that the same as the first step or generated by com-train.py in ../markerless_mouse_2/COM/train_results to predict?

3. dannce-train ../../configs/dannce_mouse_config.yaml
dannce_finetune_weights: ???
Q5: Does the dannce trained network use the finetune weight file generated by com-train.py in ../markerless_mouse_2/COM/train_results?

4. dannce-predict  ../../configs/dannce_mouse_config.yaml
Q6: which weight file is used for dannce predict model?

5. python ../../dannce/utils/makeStructuredDataNoMocap.py ./DANNCE/predict_results/save_data_AVG0.mat ../../configs/mouse22_skeleton.mat ./label3d_dannce.

**Comment by spoonsso** [comment]
Hello, sorry for the slow reply.

Q1: Yes. `com_finetune_weights` should be the path to the folder containing the pretrained weights.

Q2: `demo/markerless_mouse_1/COM/weights/` contains the pretrained rat weights, which is a reasonable starting point for mice (and what we use a starting point for training our mouse COM net). However, if you want to try some mouse weights, you can use the weights from the mouse demo, `demo/markerless_mouse_1/COM/train_results/`.

Q3: see Q2.

Q4: com-predict will use a weights file generated by com-train (i.e., after finetuning the pre-trained COM finetune weights using your labeled data). By default, it will automatically use the weights from the best epoch (defined as the lowest validation loss) without having to change any config options (these weights will be saved in `COM/train_results/`). If you want to use a different set of weights for COM prediction (e.g., the weights from the final epoch, which are placed in `COM/train_results/fullmodel_weights/fullmodel_end.hdf5`), you can add the path to the weights file in `com_predict_weights`.

Q5: No, dannce is finetuned from pretrained dannce weights. Try `demo/markerless_mouse_1/DANNCE/weight/weights.rat.MAX/`.

Q6: You will need to set `dannce_predict_model` as the path to one of the weights files generated by dannce-train, analogous to the com-predict advice in Q4. dannce-predict should normally automatically use the best epoch weights generated by dannce-train, but there is a small bug that was introduced when we added checkpoint weights callbacks, such that you will currently need to point `dannce_predict_model` directly to your target weights file. I will push the fix as soon as I am done running tests.

As for your (5), which didn't have a question. That command is used to generate a dannce prediction file that is formatted to work with https://github.com/jessedmarshall/CAPTURE_demo.

**Comment by yuan0821** [comment]
Thank you so much!!! for making such helpful suggestions. Now i could set the appropriate parameters for our lab data analysis.

**Comment by spoonsso** [comment]
In cefd782, predict-dannce will now properly use the best epoch weights in `train_results` without having to set `dannce_predict_model` manually.

**Comment by spoonsso** [comment]
> 
> 
> Thank you so much!!! for making such helpful suggestions. Now i could set the appropriate parameters for our lab data analysis.

You're welcome!
